src/libs/SearchUIUtils.ts

```diff
@@ -1446,16 +1469,30 @@ function getReportSections(
                 const reportPendingAction = reportItem?.pendingAction ?? reportItem?.pendingFields?.preview;
                 const shouldShowBlankTo = !reportItem || isOpenExpenseReport(reportItem);
                 const allActions = getActions(data, allViolations, key, currentSearch, currentAccountID, currentUserEmail, actions);
+
+                const fromDetails = transactions.length > 0 ? data.personalDetailsList[data?.[reportKey as ReportKey]?.ownerAccountID ?? CONST.DEFAULT_NUMBER_ID] : emptyPersonalDetails;
```

Comment:
The issue in the sorting is coming from here. `data.personalDetailsList` is keyed by `accountID` only, e.g. `1`,`2`, etc. But here we're trying to access it by `reportKey` which includes the Onyx prefix, e.g. `report_123`, so the lookup always fail and `fromDetails` and `formattedFrom` is always an empty string
```suggestion
                const fromDetails = data.personalDetailsList[reportItem.ownerAccountID ?? CONST.DEFAULT_NUMBER_ID];
```